Module.Introduction/Controllers/ProductsController.cs

```diff
@@ -141,7 +151,7 @@ public async Task<IActionResult> Delete(int? id)
                 return NotFound();
             }
 
-            var products = await _context.Products
+            var products = await _productsService.GetAllAsync().Result.AsQueryable()
```

Comment:
Here you are trying to get only one product (FirstOrDefaultAsync) with id but fetching all , why?